analyzers/tests/SonarAnalyzer.TestFramework/Verification/SuppressionHandler.cs

```diff
@@ -19,22 +19,32 @@
  */
 
 using System.Collections.Concurrent;
+using System.Reflection;
 using SonarAnalyzer.AnalysisContext;
 
 namespace SonarAnalyzer.TestFramework.Verification;
 
 public static class SuppressionHandler
 {
     private static readonly ConcurrentDictionary<string, int> Counters = new();
+    private static readonly PropertyInfo[] ShouldDiagnosticBeReportedProperties = LoadProperties();
```

Comment:
I considered static interfaces instead, but this problem occurs only when testing. And I didn't want to pollute the production class with a complex test-only interface, that would implement only legacy property that shouldn't even be there.